src/effects/LensFlare.tsx

```diff
@@ -0,0 +1,187 @@
+`,
+}
+
+export class LensFlareEffect extends Effect {
+  constructor({
+    blendFunction = BlendFunction.NORMAL,
+    enabled = true,
+    glareSize = 0.2,
+    lensPosition = [0.01, 0.01],
+    iResolution = [0, 0],
+    starPoints = 6,
+    flareSize = 0.01,
+    flareSpeed = 0.01,
+    flareShape = 0.01,
+    animated = true,
+    anamorphic = false,
+    colorGain = new THREE.Color(70, 70, 70),
+    lensDirtTexture = null,
+    haloScale = 0.5,
+    secondaryGhosts = true,
+    aditionalStreaks = true,
+    ghostScale = 0.0,
+    opacity = 1.0,
+    starBurst = true,
+  } = {}) {
+    super('LensFlareEffect', LensFlareShader.fragmentShader, {
+      blendFunction,
+      uniforms: new Map<string, THREE.Uniform>([
+        ['enabled', new THREE.Uniform(enabled)],
+        ['glareSize', new THREE.Uniform(glareSize)],
+        ['lensPosition', new THREE.Uniform(lensPosition)],
+        ['iTime', new THREE.Uniform(0)],
+        ['iResolution', new THREE.Uniform(iResolution)],
+        ['starPoints', new THREE.Uniform(starPoints)],
+        ['flareSize', new THREE.Uniform(flareSize)],
+        ['flareSpeed', new THREE.Uniform(flareSpeed)],
+        ['flareShape', new THREE.Uniform(flareShape)],
+        ['animated', new THREE.Uniform(animated)],
+        ['anamorphic', new THREE.Uniform(anamorphic)],
+        ['colorGain', new THREE.Uniform(colorGain)],
+        ['lensDirtTexture', new THREE.Uniform(lensDirtTexture)],
+        ['haloScale', new THREE.Uniform(haloScale)],
+        ['secondaryGhosts', new THREE.Uniform(secondaryGhosts)],
+        ['aditionalStreaks', new THREE.Uniform(aditionalStreaks)],
+        ['ghostScale', new THREE.Uniform(ghostScale)],
+        ['starBurst', new THREE.Uniform(starBurst)],
+        ['opacity', new THREE.Uniform(opacity)],
+      ]),
+    })
+  }
+
+  update(_renderer: any, _inputBuffer: any, deltaTime: number) {
+    const iTime = this.uniforms.get('iTime')
+    if (iTime) {
+      iTime.value += deltaTime
+    }
+  }
+}
+
+type LensFlareProps = ConstructorParameters<typeof LensFlareEffect>[0] & {
+  position?: THREE.Vector3
+  followMouse?: boolean
+}
+
+export const LensFlare = forwardRef<LensFlareEffect, LensFlareProps>(
+  ({ position = new THREE.Vector3(-25, 6, -60), followMouse = false, ...props }, ref) => {
```

Comment:
@ektogamat I don't know if it's possible, but maybe a more "neutral" `position` default value could be more generic: `(0,0,0)` makes sense to you ?

Reply:
It would be covered by the box default. I guess we can keep it.